components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/WADLProcessor.java

```diff
@@ -173,6 +196,19 @@ public String addWadlToRegistry(RequestContext requestContext, Resource resource
         return resource.getPath();
     }
 
+    /**
+     * This method try to delete the temporary file,
+     * If it fails it will just log a warning msg.
+     *
+     * @param file
+     * @throws IOException
+     */
+    private void delete(File file) throws IOException {
+        if (file != null && file.exists() && !file.delete()) {
+            log.warn("Failed to delete file/directory at path: " + file.getAbsolutePath());
```

Comment:
something to look at : 
https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html#deleteQuietly(java.io.File)